src/site/pages/digital/src/containers/ExprToCircuitPopup/index.tsx

```diff
@@ -0,0 +1,201 @@
+        const [userOps, setUserOps] = useState(defaultUserOps);
+
+        const inputTypes = Array.from(Inputs.keys()).map((input) =>
+            <option key={input} value={input}>{input}</option>
+        );
+    
+        const formats = Array.from(FormatMap.entries()).map((formatEntry) =>
+            <div key={formatEntry[0]}>
+                <input type="radio" id={formatEntry[0]} name="format" checked={format === formatEntry[0]} onChange={() => setFormat(formatEntry[0])} value={formatEntry[0]} />
+                <label htmlFor={formatEntry[0]}>{formatEntry[1].get("label")}</label><br/>
+            </div>
+        );
+
+        const customOps: [string, FormatLabels][] = [
+            ["AND", "&"],
+            ["OR", "|"],
+            ["XOR", "^"],
+            ["NOT", "!"],
+            ["(", "("],
+            [")", ")"],
+            ["Separator", "separator"],
+        ];
+        const customOpsInput = customOps.map((op) => 
+            <div className="exprtocircuit__popup__customOps" key={op[0]}>
+                <input title={"Enter symbol for " + op[0]} type="text" value={userOps[op[1]]} onChange={e => setUserOps({...userOps, [op[1]]: e.target.value})}/>
+                <label htmlFor={userOps[op[1]]}>Custom {op[0]}: "{userOps[op[1]]}"</label>
+            </div>
+        );
+
+        return (
+            <Popup title="Digital Expression To Circuit Generator"
+                   isOpen={(curPopup === "expr_to_circuit")}
+                   close={() => dispatch(CloseHeaderPopups())}>
+                <div className="exprtocircuit__popup">
+                    { errorMessage && <p className="exprtocircuit__popup__errorMessage">{"ERROR: " + errorMessage}</p> }
+                    <input title="Enter Circuit Expression" type="text"
+                               value={expression}
+                               placeholder="!a | (B^third)"
+                               onChange={e => setExpression(e.target.value)} />
+                    <br/>
+
+                    <div className="exprtocircuit__popup__settings">
+                        <div>
+                            <h3>Notation</h3>
+                            {formats}
+                            <div>
+                                <input type="radio" name="format" checked={format === "custom"} onChange={() => setFormat("custom")} value={"custom"} />
+                                <label htmlFor={"custom"}>Custom</label><br/>
+                            </div>
+                            {
+                                format === "custom" &&
+                                <div>
+                                    {customOpsInput}
+                                </div>
+                            }
+                        </div>
+
+                        <div>
+                            <h3>Options</h3>
+                            <input onChange={() => setIsIC(!isIC)} checked={isIC} type="checkbox" id="isIC" name="isIC" />
```

Comment:
nit: id should be first property, followed by `type` then `name`, `onChange` should be last